utils/frame/benchmarking-cli/src/storage/write.rs

```diff
@@ -62,49 +68,102 @@ impl StorageCmd {
+			if self.params.include_child && k.starts_with(DEFAULT_CHILD_STORAGE_KEY_PREFIX) {
+				child_tries.push(k.clone());
+			}
+
 			'retry: loop {
 				let mut new_v = vec![0; original_v.len()];
-				// Create a random value to overwrite with.
-				// NOTE: We use a possibly higher entropy than the original value,
-				// could be improved but acts as an over-estimation which is fine for now.
+				// 	// Create a random value to overwrite with.
+				// 	// NOTE: We use a possibly higher entropy than the original value,
+				// 	// could be improved but acts as an over-estimation which is fine for now.
 				rng.fill_bytes(&mut new_v[..]);
-				let new_kv = vec![(k.as_ref(), Some(new_v.as_ref()))];
-				let (_, mut stx) = trie.storage_root(new_kv.iter().cloned(), self.state_version());
-				for (mut k, (_, rc)) in stx.drain().into_iter() {
-					if rc > 0 {
-						db.sanitize_key(&mut k);
-						if db.get(state_col, &k).is_some() {
-							trace!("Benchmark-store key creation: Key collision detected, retry");
-							continue 'retry
-						}
-					}
+				match check_new_value::<Block>(
+					db.clone(),
+					&trie,
+					&k.to_vec(),
+					&new_v,
+					self.state_version(),
+					state_col,
+					None,
+				) {
+					true => {
+						*original_v = new_v;
+						break
+					},
+					false => continue 'retry,
 				}
-				*original_v = new_v;
-				break
 			}
 		}
 
 		info!("Writing {} keys", kvs.len());
 		// Write each value in one commit.
 		for (k, new_v) in kvs.iter() {
-			// Interesting part here:
-			let start = Instant::now();
-			// Create a TX that will modify the Trie in the DB and
-			// calculate the root hash of the Trie after the modification.
-			let replace = vec![(k.as_ref(), Some(new_v.as_ref()))];
-			let (_, stx) = trie.storage_root(replace.iter().cloned(), self.state_version());
-			// Only the keep the insertions, since we do not want to benchmark pruning.
-			let tx = convert_tx::<Block>(db.clone(), stx.clone(), false, state_col);
-			db.commit(tx).map_err(|e| format!("Writing to the Database: {}", e))?;
-			record.append(new_v.len(), start.elapsed())?;
-
-			// Now undo the changes by removing what was added.
-			let tx = convert_tx::<Block>(db.clone(), stx.clone(), true, state_col);
-			db.commit(tx).map_err(|e| format!("Writing to the Database: {}", e))?;
+			let res = calculate_bench::<Block>(
+				db.clone(),
+				&trie,
+				k.to_vec(),
+				new_v.to_vec(),
+				self.state_version(),
+				state_col,
+				None,
+			)?;
+			record.append(res.0, res.1)?;
+		}
+
+		if self.params.include_child {
+			info!("Writing {} child trees", child_tries.len());
+
+			for key in child_tries {
+				let trie_id = key
+					.strip_prefix(DEFAULT_CHILD_STORAGE_KEY_PREFIX)
+					.expect("Checked above to exist");
+				let info = ChildInfo::new_default(trie_id);
+				let child_keys =
+					client.child_storage_keys_iter(&block, info.clone(), None, None)?;
+
+				for ck in child_keys {
+					if let Some(original_v) = client
```

Comment:
I wish there was a way to get all key, value pairs inside a child tree but was not able to find anything for that, that's why here we are reading the value for each child key